src/routes/transactions/mappers/queued-items/queued-items.mapper.ts

```diff
@@ -114,24 +114,16 @@ export class QueuedItemsMapper {
 
   /**
    * Divides an array of transactions in groups with the same nonce.
+   * Each TransactionGroup will contain at least one transaction.
    * @param transactions transactions with potentially different nonces.
    * @returns Array<TransactionGroup> in which each group of transactions has a different nonce.
    */
   private groupByNonce(transactions: Transaction[]): TransactionGroup[] {
-    const groups = transactions.reduce((result, transaction) => {
-      const executionInfo = transaction?.executionInfo as MultisigExecutionInfo;
-      return {
-        ...result,
-        [executionInfo['nonce']]: [
-          ...(result[executionInfo['nonce']] || []),
-          transaction,
-        ],
-      };
-    }, {});
-
-    return Object.entries(groups).map((group) => ({
-      nonce: Number(group[0]),
-      transactions: group[1] as Transaction[],
-    }));
+    return Object.entries(groupBy(transactions, 'executionInfo.nonce')).map(
```

Comment:
I think working with the route entity at this level has some downsides. You loose some domain related types that can be useful when mapping types.

In this example `nonce` is a `string` even though we end up mapping it to a `number`.

Why not work with the domain entity and map to the final route entity at the end?

If going with the domain option we can explicitly say that we expect a `MultisigTransaction` and not any other transaction. As it stands now `executionInfo.nonce` is not available for every  `Transaction`.

Reply:
I changed this to use `MultisigTransaction` into `QueuedItemsMapper`. It also gave me the opportunity to refactor the code a bit and merge `getAsFirstTransaction` and `getAsNonFirstTransactions` into a hopefully simpler `getConflictType` which is called for each transaction. I think it's easier to follow the code this way.